### ISSUE TITLE
[FP]: snakeyaml-1.32.jar

### ISSUE THREAD

**Issue by whydoievenneedthis** [dependency-check/DependencyCheck#4919]
### Package URl

pkg:maven/org.yaml/snakeyaml@1.32

### CPE

`cpe:2.3:a:snakeyaml_project:snakeyaml:1.32:*:*:*:*:*:*:*`

### CVE

CVE-2022-38752

### ODC Integration

{"label"=>"Gradle Plugin"}

### ODC Version

7.2.1

### Description

https://nvd.nist.gov/vuln/detail/CVE-2022-38752
![image](https://user-images.githubusercontent.com/87586203/194941149-1340d0c6-2528-4533-bfaa-5b9958f599ac.png)

Btw. it also trips up on 1.33:
```
snakeyaml-1.33.jar (pkg:maven/org.yaml/snakeyaml@1.33, cpe:2.3:a:snakeyaml_project:snakeyaml:1.33:*:*:*:*:*:*:*, cpe:2.3:a:yaml_project:yaml:1.33:*:*:*:*:*:*:*) : CVE-2022-38752
```

**Comment by aikebah** [comment]
NVD is not the only source. If you check the generated report you're likely to find the (OSSINDEX) suffix to the CVE number.

See also https://github.com/OSSIndex/vulns/issues/328

**Comment by aikebah** [comment]
Based on the history in the issues my impression is that the Sonatype researchers are likely correct that it is not fixed in the newer versions, but newer versions appear to offer some handles to mitigate some (but not all) occurrences of the issue.

The main response appears to be an 'I have to allow it as the YAML spec requires it' from the developer ignoring the 'If you want to strictly follow the YAML spec you should use a different datastructure for the deserialization' remarks of the commenters.

**Comment by mprins** [comment]
duplicates #4839

**Comment by chadlwilson** [comment]
I suspect one might be granting the Sonatype researchers a bit too much credit here. :-) There was no released fix at time that the issue was disclosed, now they need to go back and update. I've seen zero engagement on issues raised via their GitHub (but might be missing something) - last issue closed was 24 Aug. Even NIST NVD do better than that with community engagement.

I personally asked NIST NVD to review their affected versions and got them to update, so they deem it fixed. I made sure I understand what was changed before requesting this so evidence could be provided - if you look at my last comment at https://bitbucket.org/snakeyaml/snakeyaml/issues/531/stackoverflow-oss-fuzz-47081 The TL;DR is that the "default" configuration already disables recursive keys. Ossfuzz found a case where those recursive keys were not blocked as intended. That specific case was also blocked with a code change. There *may* be other cases missed, and so there was some debate about removing support for recursive keys entirely; and that's where it goes back to a spec argument.

But I agree the issue discussion was very confusing to follow and went around in circles in various places.

Just wondering, is possible for us to close all these issues as "not planned" where they are to do with individual CVE false positives and effectively upstream data? I'm not sure what the value is of clogging up the DependencyCheck issues log with FPs that aren't to do with bugs/weaknesses of the library (i.e bad CPE matches and such) and for which there is no code/config change to make in DependencyCheck itself. It makes it hard to find things on the issue backlog and appears to not stop people opening duplicates regardless of open/closed status.
